ui/lib/pki/addon/components/key-usage.ts

```diff
@@ -0,0 +1,82 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+/**
+ * @module PkiKeyUsage
+ * PkiKeyUsage components are used to build out the toggle options for PKI's role create/update key_usage, ext_key_usage and ext_key_usage_oids model params.
+ * Instead of having the user search on the following goLang pages for these options we present them in checkbox form and manually add them to the params as an array of strings.
+ * key_usage options: https://pkg.go.dev/crypto/x509#KeyUsage
+ * ext_key_usage options (not all are include on purpose): https://pkg.go.dev/crypto/x509#ExtKeyUsage
+ * @example
+ * ```js
+ * <PkiKeyUsage @model={@model} @group={group}/>
+ * ```
+ * @param {class} model - The pki/pki-role-engine model.
+ * @param {string} group - The name of the group created in the model. In this case, it's the "Key usage" group.
+ */
+
+interface Field {
+  key: string;
+  label: string;
+}
+
+const KEY_USAGE_FIELDS: Field[] = [
```

Comment:
Not for now, but just as a spitballing note - wondering if these could live in a pki helper if we have any other fields that could/will be stored as `const`s. Otherwise - do we need to define these here? Or can we directly set these values on `keyUsageFields` and `extKeyUsageFields` below?